src/main/java/org/sagebionetworks/web/client/widget/docker/DockerRepoWidget.java

```diff
@@ -1,44 +1,88 @@
 package org.sagebionetworks.web.client.widget.docker;
 
+import java.util.Map;
+
 import org.sagebionetworks.repo.model.EntityBundle;
+import org.sagebionetworks.repo.model.ObjectType;
+import org.sagebionetworks.web.client.DisplayUtils;
+import org.sagebionetworks.web.client.events.EntityUpdatedEvent;
+import org.sagebionetworks.web.client.events.EntityUpdatedHandler;
+import org.sagebionetworks.web.client.utils.CallbackP;
 import org.sagebionetworks.web.client.widget.entity.WikiPageWidget;
 import org.sagebionetworks.web.client.widget.entity.controller.PreflightController;
 import org.sagebionetworks.web.client.widget.entity.controller.SynapseAlert;
-import org.sagebionetworks.web.client.widget.entity.tabs.DockerTab;
+import org.sagebionetworks.web.client.widget.provenance.ProvenanceWidget;
+import org.sagebionetworks.web.shared.WidgetConstants;
+import org.sagebionetworks.web.shared.WikiPageKey;
 
 import com.google.gwt.user.client.ui.Widget;
 import com.google.inject.Inject;
 
 public class DockerRepoWidget implements DockerRepoWidgetView.Presenter{
+	public static final String DOCKER_PULL_COMMAND = "docker pull ";
+
 	private PreflightController preflightController;
 	private DockerRepoWidgetView view;
 	private SynapseAlert synAlert;
 	private WikiPageWidget wikiPageWidget;
+	private ProvenanceWidget provWidget;
+	private EntityUpdatedHandler handler;
 
 	@Inject
 	public DockerRepoWidget(
 			PreflightController preflightController,
```

Comment:
While working on ActionMenu, if this is not used, need to be remove. Same with synAlert.